### ISSUE TITLE
Location tracking poorly when device is locked

### ISSUE THREAD

**Issue by brneas** [transistorsoft/flutter_background_geolocation#432]
## Your Environment
* Plugin version: 1.10.1
* Platform: iOS 14.1 Android 10
* OS version: Windows 10. macOS Catalina 10.15.7
* Device manufacturer / model: Samsung Note 10+. iPhone SE 2nd generation
* Flutter info (`flutter doctor`):

[√] Flutter (Channel beta, 1.20.2, on Microsoft Windows [Version 10.0.18363.1139], locale en-US)
    • Flutter version 1.20.2 at C:\Src\flutter
    • Framework revision bbfbf1770c (2 months ago), 2020-08-13 08:33:09 -0700
    • Engine revision 9d5b21729f
    • Dart version 2.9.1

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.1)
    • Android SDK at C:\Users\USER\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.1
    • Java binary at: C:\Program Files (x86)\Common Files\Oracle\Java\javapath\java.exe
    • Java version Java(TM) SE Runtime Environment (build 1.8.0_261-b12)
    • All Android licenses accepted.
* Plugin config:
```dart <-- Syntax highlighting: DO NOT REMOVE -->
 if(Platform.isAndroid){
   bg.BackgroundGeolocation.ready(
     bg.Config(
       desiredAccuracy: bg.Config.DESIRED_ACCURACY_HIGH,
       distanceFilter: 10.0,
       stopOnTerminate: false,
       startOnBoot: true,
       debug: true,
       logLevel: bg.Config.LOG_LEVEL_VERBOSE,
       notification: bg.Notification(
         title: 'App',
         text: 'App is tracking your location',
         smallIcon: 'images/logo.png',
         largeIcon: 'images/logo.png',
       ),
     ),
   ).then((bg.State state) {
     if (!state.enabled) {
       bg.BackgroundGeolocation.start();
     }
   });
 }
 else{
   bg.BackgroundGeolocation.ready(
     bg.Config(
         desiredAccuracy: bg.Config.DESIRED_ACCURACY_NAVIGATION,
       preventSuspend: true,
       heartbeatInterval: 10,
       startOnBoot: true,
       stationaryRadius: 10.0,
     ),
   ).then((bg.State state) {
     if (!state.enabled) {
       bg.BackgroundGeolocation.start();
     }
   });
 }
}
```

## Expected Behavior
I expect that when my device is locked that my location will still be tracked/logged.

## Actual Behavior
When my device is locked my location logs are sporadic or incorrect. Logs occur less. See examples below

## Steps to Reproduce

1. Lock device while attempting to track location

## Context

I am trying to build an application to track employees in the field. The application needs to work when the device is locked but when my testing devices are locked my location is not tracked correctly.
Below is an image of my results. I walked around the border of about half an acre of land.
A1: Android with phone unlocked and app open
A2: Android with phone locked
A3: iPhone with phone unlocked and app open
A4: iPhone with phone locked

![Example image](https://i.imgur.com/gTYK6CS.png)

## Debug logs

<details>
	<summary>Logs</summary>

``` <!-- syntax-highligting:  DO NOT REMOVE -->
I was using physical devices... I'm not sure how to get logs off of them but if you need them I'll try to figure it out.
```

</details>

**Comment by christocracy** [comment]
It makes no difference whether app is in background, device locked or app terminated.

- See http://dontkillmyapp.com.
- is your battery low?

**Comment by brneas** [comment]
Thanks for that information.

On Android I have checked that my app is not sleeping, I disabled "Put unused apps to sleep", and I went into my phone settings and disabled "Optimize Battery usage" on my app.
After making those changes the location tracking for my app did not get any better when the device was locked.

My battery is not low. On both devices it is over 60%.

**Comment by christocracy** [comment]
Go for a long field-test.  I test on 11 Android devices almost daily.  I do not touch these devices.  I keep the sample app running on them for weeks at a time, place them in my backpack and go.  When I return, I plugin them back into power.  They do not fail.

![](https://dl.dropbox.com/s/svvjzv69y8o9byw/Screen%20Shot%202020-10-23%20at%202.13.58%20PM.png?dl=1)

**Comment by christocracy** [comment]
You'll have a much better time testing using the demo-server at `tracker.transistorsoft.com`.  It's very easy to configure -- see API docs [TransistorAuthorizationToken](https://pub.dev/documentation/flutter_background_geolocation/latest/flt_background_geolocation/TransistorAuthorizationToken-class.html)

**Comment by brneas** [comment]
I went on a long field test. I drove around about 5 miles with the same settings from my original issue. Android works a little better but on the second half of my trip you can see that the tracking became more sporadic. 
![Android Long Field Test](https://user-images.githubusercontent.com/9089998/97042796-901ee980-153f-11eb-84f9-4f126cc3cab5.PNG)

I took my iPhone with me as well and it did not do any better on the long field test. It tracked points when I first started tracking but as soon as I locked the device it stopped working.
![iPhone Long Field Test](https://user-images.githubusercontent.com/9089998/97042797-90b78000-153f-11eb-8ecc-31216f7f5b9c.PNG)

I already have an API setup that I feel comfortable testing on but if it would make things easier for you I can switch to using your demo server.

**Comment by christocracy** [comment]
I suggest you install the `/example` app in this repo and compare performance to your own app.  Your problem is either:
- due to the environment
- device
- your code.

iOS performances has been remarkable for over 6 years now, tested all over the world.

**Comment by christocracy** [comment]
>your code.

You've not provided `stopOnTerminate` to your iOS config.

**Comment by christocracy** [comment]
Closed with no explanation?

**Comment by brneas** [comment]
I installed the /example app from this repo onto my iPhone and Android device. I verified that worked.

I nuked my code and borrowed some of the /example code to get the result that I wanted. I then verified that worked.

I was unable to find exactly what was causing the iPhone issues if I had to guess I think the add did not have Allow Location Access marked as always. For some reason I did not get the permission prompt until trying the /example code.

I'm currently working on the issue of why the markers on my map fall off about 3 miles after usage. I'm thinking this is a result of the API being flooded with requests and just dropping them. I'm in the process of implementing the HTTPEvent since that is better. 
Currently the HTTPEvent is posting an "undefined" request.body which doesn't make sense to me... and my Dart DevTools are not showing any network response/requests which makes even less sense but that doesn't have anything to do with this issue so everything is resolved.

Thanks!
